bitcoin/src/dogecoin/mod.rs

```diff
@@ -283,4 +287,124 @@ mod tests {
         let got = Target::from_compact(CompactTarget::from_consensus(bits));
         assert_eq!(got, want)
     }
+
+    #[test]
+    fn compact_target_from_upwards_difficulty_adjustment() {
+        let params = Params::new(Network::Dogecoin);
+        let starting_bits = CompactTarget::from_consensus(0x1e0ffff0); // Genesis compact target on Mainnet
+        let start_time: u64 = 1386325540; // Genesis block unix time
+        let end_time: u64 = 1386475638; // Block 239 unix time
+        let timespan = end_time - start_time; // Slower than expected (150,098 seconds diff)
+        let adjustment = CompactTarget::from_next_work_required_dogecoin(starting_bits, timespan, &params);
```

Comment:
```suggestion
        let adjustment = CompactTarget::from_next_work_required_dogecoin(starting_bits, timespan, Network::Dogecoin);
```

Similarly in all places where `AsRef<Params>` is expected, you can directly pass `Network`.

Reply:
Yes that's a good point. Also, I based this test from bitcoin's which uses `params`, so for consistency we could as well leave `params` as it is WDYT?

Reply:
Thanks for the explanation. No problem!